packages/amplify-provider-awscloudformation/src/root-stack-builder/root-stack-builder.ts

```diff
@@ -0,0 +1,252 @@
+      value: cdk.Fn.ref('AWS::StackId'),
+      exportName: cdk.Fn.sub('${AWS::StackName}-StackId'),
+    });
+
+    cfnOutputs.authRoleArn = new cdk.CfnOutput(this, 'AuthRoleArn', {
+      value: cdk.Fn.getAtt('AuthRole', 'Arn').toString(),
+    });
+
+    cfnOutputs.unauthRoleArn = new cdk.CfnOutput(this, 'UnauthRoleArn', {
+      value: cdk.Fn.getAtt('UnauthRole', 'Arn').toString(),
+    });
+    this.templateObj = {};
+    this.templateObj['Parameters'] = { ...CfnParameters };
+    this.templateObj['Outputs'] = { ...cfnOutputs };
+    this.templateObj.Resources = { ...CfnResources };
+  }
+
+  toCloudFormation() {
+    prepareApp(this);
+    return this._toCloudFormation();
+  }
+}
+
+export enum CommandType {
+  'PUSH',
+  'INIT',
+}
+
+type RootStackOptions = {
+  rootStackFileName: string;
+  event: CommandType;
+  overrideDir?: string;
+};
+
+/**
+ * additional class to merge CFN parameters and CFN outputs as cdk doesnt allow same logical ID of constructs in same stack
+ */
+export class AmplifyRootStackOutputs extends cdk.Stack {
+  public templateObj: AmplifyRootStackTemplateProps;
+  constructor(scope: cdk.Construct, id: string) {
+    super(scope, id);
+    const cfnOutputs: { [k: string]: cdk.CfnOutput } = {};
+    cfnOutputs.deploymentBucketName = new cdk.CfnOutput(this, 'DeploymentBucketName', {
+      description: 'CloudFormation provider root stack deployment bucket name',
+      value: cdk.Fn.ref('DeploymentBucketName'),
+      exportName: cdk.Fn.sub('${AWS::StackName}-DeploymentBucketName'),
+    });
+
+    cfnOutputs.authRoleName = new cdk.CfnOutput(this, 'AuthRoleName', {
+      value: cdk.Fn.ref('AuthRoleName'),
+    });
+
+    cfnOutputs.unauthRoleName = new cdk.CfnOutput(this, 'UnauthRoleName', {
+      value: cdk.Fn.ref('UnauthRoleName'),
+    });
+    this.templateObj = {};
+    this.templateObj['Outputs'] = { ...cfnOutputs };
+  }
+
+  toCloudFormation() {
+    prepareApp(this);
+    return this._toCloudFormation();
+  }
+}
+
+export interface ResourceConfig {
+  resourceName?: string;
+  category: string;
+  stackFileName: string;
+}
+
+export interface RootStackTransformOptions {
+  //inputParser: IInputParser[];  // not needed for root stack
+  resourceConfig?: ResourceConfig;
+}
+
+export interface DeploymentOptions {
+  templateStack: Template;
+}
+
+export class AmplifyRootStackTransform {
+  private _rootTemplateObj: AmplifyRootStackTemplateProps; // Props to modify Root stack data
+  private _resourceConfig: ResourceConfig; // Config about resource to build
+  private _rootStackOptions: RootStackOptions; // options to help generate  cfn template
+  private _command: CommandType;
+  constructor(options: RootStackTransformOptions, command: CommandType) {
+    this._resourceConfig = options.resourceConfig;
+    this._command = command;
+  }
+
+  public async transform(): Promise<Template> {
+    // parse Input data
+    this._rootStackOptions = await this.getInput(); // get RootStackOPtions from cli-inputs.json
```

Comment:
`getInput` does not seems to read cli-input.json as the comment suggests it is more like `getRootStackOptions` or `getStackOptions` ?